localstack-core/localstack/services/ec2/patches.py

```diff
@@ -109,11 +110,29 @@ def ec2_create_vpc(
                     vpc_id=vpc_id,
                 )
 
+            # Delete route table if only main route table remains.
+            route_tables = self.describe_route_tables(filters={"vpc-id": vpc_id})  # type: ignore[attr-defined]
+            if len(route_tables) > 1:
+                raise DependencyViolationError(
+                    f"The vpc {vpc_id} has dependencies and cannot be deleted."
```

Comment:
Why the VPC? Shouldn't this be the route table?

Reply:
Good catch. Simplified the deletion of the route tables.